src/main.rs

```diff
@@ -64,6 +70,11 @@ async fn amain(mut leds: Leds, mut wifi: AsyncWifi<EspWifi<'static>>) {
                 log::error!("ZAP: Printer error: {e}");
             }
         }
+        if matches!(button_switch.get_level(), Level::High) {
```

Comment:
You should have a state variable here that doesnt block but doesnt print multiple times when the button is pressed. i.e., if you press the button it should only print once, then wait for the button to not be pressed instead of being able to send another receipt request.

Reply:
Err I thought we were magically protected by that by the fact that the code in this loop only seems to run once every second? My understanding was that every second it's checking the state of the pins, if the button is pressed then it sends a print command, then blocks and checks again after 1 second. If this is how it works, why would it be possible to send multiple print commands? Other than holding the button for 2+ seconds

Reply:
I completely forgot about that, but that actually presents a different problem: If the user pushes the button while this is sleeping, it will be ignored. You should decrease the sleep delay and also add my suggestion since it will then become necessary.